### DISCUSSION TITLE
Extending native migration capabilities

### DISCUSSION THREAD

**Discussion by eyecatchup** [directus/directus#11367] [Feature Requests]
> Hope this is the right starting point. Contributing guidelines sent me here.

**Background:** There's a merged PR, which added the [native _schema_ migration](https://github.com/directus/directus/pull/7939) feature to the CLI, back in September 2021. It kicked off a discussion about extending the feature and what should be the scope of it.

@benhaynes [summarized](https://github.com/directus/directus/pull/7939#issuecomment-916416086) 

> I also agree that from a user's perspective, all of these things will likely happen at once: during "migration". So it makes sense to aggregate all of this under one migration umbrella. Again, not technical, but I see this as some sort of migration CLI/endpoint/etc that has options or flags for schema, system config, content. Behind the scenes, the code can be organized to separate schema and content, but agree that it would make sense to expose the functionality through one platform function.

[and](https://github.com/directus/directus/pull/7939#issuecomment-916379609) 

> [..] that the goal of this feature should be to export everything (schema, system, content, etc).

Adding some loose proposals of implementation, he had discussed with @rijkvanzanten.

---

Now, because I wasn't able to find any follow ups on this discussion and it is a rather important feature for my company's needs, I'd be happy to contribute on this feature. So, this issue is meant to

- ask if there is any ongoing (private) development on this 
- and, if not, to
  - offer my help implementing the feature expansion
  - start a dedicated discussion thread on the feature expansion

**Comment by rijkvanzanten** [comment]
Hi @eyecatchup!

Love it. As of right now, there aren't any in-works PRs. Mind shooting me a DM on [Discord](https://directus.chat)? Little easier to briefly discuss on there what you're thinking about, and how we can proceed 🙂

**Comment by OnescuRadu** [comment]
Hi @eyecatchup & @rijkvanzanten!
Do you have any updates on this one?

Would be great to be able to migrate permissions and user roles.

**Reply by rijkvanzanten** [reply]
No new updates 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
